controllers/webhook/runcompletion_event_handler.go

```diff
@@ -0,0 +1,10 @@
+package webhook
+
+import (
+	"context"
+	"github.com/sky-uk/kfp-operator/argo/common"
+)
+
+type RunCompletionEventHandler interface {
+	handle(ctx context.Context, ed common.RunCompletionEvent) error
```

Comment:
why `ed` and not just `event`?
```suggestion
	handle(ctx context.Context, event common.RunCompletionEvent) error
```